File: 3300-minimum-element-after-replacement-with-digit-sum/solution.py
class Solution:
    def minElement(self, nums: list[int]) -> int:
        def get_digit_sum(n: int) -> int:
            digit_sum = 0
            while n > 0:
                digit_sum += n % 10
                n //= 10
            return digit_sum
        
        # Generator expression to find the minimum digit sum efficiently
        return min(get_digit_sum(num) for num in nums)

File: 3300-minimum-element-after-replacement-with-digit-sum/solution.cpp
#include <vector>
#include <algorithm>
#include <climits>

class Solution {
private:
    // Helper function to calculate the sum of digits of a number
    int getDigitSum(int n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }

public:
    int minElement(std::vector<int>& nums) {
        int minVal = INT_MAX;
        for (int num : nums) {
            minVal = std::min(minVal, getDigitSum(num));
        }
        return minVal;
    }
};